File: Medium/Array/1288. Remove Covered Intervals.py
# 1288. Remove Covered Intervals
# nlog(n)
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:

        intervals.sort()
        n = len(intervals)
        start = intervals[0][0]
        end = intervals[0][1]
        count = 0

        for i in range(1,n,1):
            x = intervals[i][0]
            y = intervals[i][1]
            if(x >= start and y <= end):
                count+=1
                continue
            elif(x == start and y > end):
                count+=1
            
            start = x
            end = y
        return n - count

        

File: Medium/Array/1288. Remove Covered Intervals.cpp
// 1288. Remove Covered Intervals 
// O(n^2)
class Solution {
public:
    int removeCoveredIntervals(vector<vector<int>>& intervals) {

        int n = intervals.size();
        vector<bool>remove(n,false);
        int count = 0;
        sort(intervals.begin(),intervals.end());

        for(int i = 0; i<n;i++){
            if(remove[i]) continue;
            for(int j = i+1; j<n;j++){
                if(remove[j]) continue;
                int p = intervals[i][0];
                int q = intervals[i][1];
                int x = intervals[j][0];
                int y = intervals[j][1];
                if(x >= p && y<=q){
                    remove[j] = true;
                    count++;
                }
                else if(p>=x && q<=y){
                    count++;
                    break;
                }
            }
        }
        return n-count;
        
    }
};
